phpBB/phpbb/controller/helper.php

```diff
@@ -184,6 +184,8 @@ public function route($route, array $params = array(), $is_amp = true, $session_
 	* @param string $message The error message
 	* @param int $code The error code (e.g. 404, 500, 503, etc.)
 	* @return Response A Response instance
+	*
+	* @deprecated 3.1.3 (To be removed: 3.3.0) Use exceptions instead.
```

Comment:
Using helper->error is dictated by extension policies (https://www.phpbb.com/extensions/rules-and-policies/validation-policy/#php #12) so the way this PR intends to replace that with these exceptions should be documented in Ext dev or writers forums with specific examples.

PS: I assume these won't replace trigger_error's current functionality in the ACP yet?